packages/shared/src/hooks/prompt/useSmartPrompt.ts

```diff
@@ -93,6 +96,9 @@ export const useSmartPrompt = ({
           case UseChatMessageType.Completed: {
             setSearchQuery({ completedAt: new Date() });
             sourceRef.current?.close();
+            if (lastPrompt !== prompt.id) {
+              updateFlag('lastPrompt', prompt.id);
```

Comment:
we can introduce `updateFlag` remote which only updates remote data to avoid live re-sort